### PR TITLE
[IMP] views: setting view

### PR DESCRIPTION
**Pull request by jpp-odoo**
This commit, add the documentation of the new DSL created exclusively for the settings.

task-id: 3081367

### CODE REVIEW

File: content/developer/reference/backend/views.rst [hunk 1]
```diff
@@ -2527,6 +2527,132 @@ For instance if ``foo`` and ``bar`` refer to two groupbys
...
+-------------------
+
+As seen in :ref:`Customizing an existing view <reference/js_class>` we can create a custom version
+of a generic view. Here we will present some of these views:
+
+Settings View
+~~~~~~~~~~~~~
+
+The settings view is a customization of the form view. It's used to centralize all the settings of Odoo.
+This view differs from a generic form view because it has a search panel and a sidebar.
```
Comment (aab-odoo) [inline content/developer/reference/backend/views.rst:2540]: search bar?

File: content/developer/reference/backend/views.rst [hunk 2]
```diff
@@ -2527,6 +2527,132 @@ For instance if ``foo`` and ``bar`` refer to two groupbys
 
 has the effect to activate first ``bar`` then ``foo``.
 
+View customizations
+-------------------
+
+As seen in :ref:`Customizing an existing view <reference/js_class>` we can create a custom version
+of a generic view. Here we will present some of these views:
+
+Settings View
```
Comment (aab-odoo) [inline content/developer/reference/backend/views.rst:2536]: I think I'd place this section inside the Form View section (thus removing the above paragraph)

File: content/developer/reference/backend/views.rst [hunk 3]
```diff
@@ -2527,6 +2527,132 @@ For instance if ``foo`` and ``bar`` refer to two groupbys
...
+~~~~~~~~~~~~~
+
+The settings view is a customization of the form view. It's used to centralize all the settings of Odoo.
+This view differs from a generic form view because it has a search panel and a sidebar.
+
+To avoid complecity, simplify and standardize the settings archs, a small DSL exclusively for the
+settings was created. This new DSL introduces 3 tags: ``app``, ``block`` and ``setting``.
```
Comment (aab-odoo) [inline content/developer/reference/backend/views.rst:2543]:
```suggestion
The settings form view accepts three additional tags: ``app``, ``block`` and ``setting``.
```

File: content/developer/reference/backend/views.rst [hunk 4]
```diff
@@ -2527,6 +2527,132 @@ For instance if ``foo`` and ``bar`` refer to two groupbys
...
+    <block title="Title of group Bar">
+    ...
+    </block>
+
+  Parameters:
+
+  .. rst-class:: o-definition-list
+
+  ``title`` (optional)
+    The title of the block of settings (the old h2), you can perform research on its text.
```
Comment (aab-odoo) [inline content/developer/reference/backend/views.rst:2586]: in the doc, there should be no notion of "old" (unlike in the commit, where you explain what changes).

**Review by aab-odoo — APPROVED** [review]

File: content/developer/reference/backend/views.rst [hunk 5]
```diff
@@ -1014,6 +1014,124 @@ Generic structure
 
 .. todo:: widgets?
 
+Settings Form View
+------------------
```
Comment (AntoineVDV) [inline content/developer/reference/backend/views.rst:1018]: Shouldn't this section be a sub-section of Form?
```suggestion
~~~~~~~~~~~~~~~~~~
```
The title could be **Settings components** to be consistent with the other sub-sections, for example.

File: content/developer/reference/backend/views.rst [hunk 6]
```diff
@@ -1014,6 +1014,124 @@ Generic structure
...
+  ``title`` (optional)
+    The text used as tooltip.
+  ``help`` (optional)
+    The help/description of the setting. This text is displayed just below the setting label (with classname ``text-muted``).
+  ``company_dependent`` (optional)
+    If this attribute is set to "1" an icon is displayed next to the setting label to explicit
+    that this setting is company-specific.
+  ``documentation`` (optional)
+    If this attribute is set, an icon is added next to the setting label, this icon is a link to the documentation.
+    Note that you can use relative or absolute path. The relative path is relative to ``https://www.odoo.com/documentation/server_version``,
```
Comment (AntoineVDV) [inline content/developer/reference/backend/views.rst:1108]: Nitpicking.
```suggestion
    Note that you can use relative or absolute path. The relative path is relative to ``https://www.odoo.com/documentation/<server_version>``,
```

File: content/developer/reference/backend/views.rst [hunk 7]
```diff
@@ -1014,6 +1014,124 @@ Generic structure
...
+            <setting string="This is Big BAR" company_specific="1">
+                <field name="bar"/>
+            </setting>
+        </block>
+        <block title="Title of group Foo">
+            <setting string="Personalize setting" help="this is full personalize setting">
+                <div>This is a different setting</div>
+            </setting>
+        </block>
+    </app>
```
Comment (AntoineVDV) [inline content/developer/reference/backend/views.rst:1133]: Use the `example` directive.
```rst
.. example::

   .. code-block:: XML

       <app string="CRM" name="CRM">
           <setting type="header" string="Foo">
            ...
```

File: content/developer/reference/backend/views.rst (hunk 5 above)
Comment (jpp-odoo) [reply content/developer/reference/backend/views.rst:1018]: The settings view, is not a component of the form view. It's a jsClass, a customization of the form view, that have a search bar (form view don't have one), a sidebar, and had its own components. It's for these reasons that we put-it as an independent view.

**Review by aab-odoo — APPROVED** [review]
robodoo r+

**Comment by jpp-odoo** [comment]
@robodoo r+

**Pull request closed without merging**